### PR TITLE
Gh 364 add support for transforms

### LINKED ISSUE #364: Add a spec update method that will add the spec changes to the spec diff annotation and also to the ingress spec
Add a method within the ingress package that will allow us to encapsulate the udpdates we make to the spec of an ingress. This method should add the  update to the spec diif annotation ```ClusterSpecDiffAnnotationPrefix``` but also currently add the change to the ingress spec. The main changes we make to spec currently are
- replacing /adding the host(s) in the ingress
- replacing/adding the tls section

- [x] New method(s) in place and any update the ingress spec done via that method
- [ ] E2e tests updated to expect the annotation to be set
- [ ] Unit tests in place for new method

### PR DESCRIPTION
**Pull request by maleck13**
Closes #364 #399 #459 

## Description of Changes

remove advanced scheduling flag and use and `tmcEnabled()` method
use the spec diff annotations to set the spec of the downstream ingress
set the lb status of the upstream ingress to be the glbc provided dns name
remove the custom domains enabled flag
remove the tls enabled flag

## Release and Testing
There are e2e tests in place for ingress and the smoke tests
There is no test for routes yet as that is part of another issue being progressed

### CODE REVIEW

File: samples/echo-service/echo.yaml [hunk 1]
```diff
@@ -39,8 +39,8 @@ metadata:
   name: echo
 spec:
   rules:
-    - host:
-      http:
+    - host: craigbrookes.com
```
Comment (maleck13) [inline samples/echo-service/echo.yaml:42]: remove

File: pkg/traffic/host_test.go [hunk 2]
```diff
@@ -550,9 +505,10 @@ func TestProcessCustomHostValidation(t *testing.T) {
 			for _, expectedTLS := range testCase.expectedTLS {
 				foundExpectedTLS := false
 				for _, tls := range ingressAccessor.Spec.TLS {
+					fmt.Println("ingress tls ", tls)
```
Comment (philbrookes) [inline pkg/traffic/host_test.go:508]: remove this.

File: pkg/traffic/host_test.go [hunk 3]
```diff
@@ -550,9 +505,10 @@ func TestProcessCustomHostValidation(t *testing.T) {
...
 					if equality.Semantic.DeepEqual(expectedTLS, tls) {
 						foundExpectedTLS = true
-						break
+						//break
```
Comment (philbrookes) [inline pkg/traffic/host_test.go:511]: and this.

File: samples/echo-service/echo.yaml [hunk 4]
```diff
@@ -39,8 +39,8 @@ metadata:
   name: echo
 spec:
   rules:
-    - host: host.whatever.com
-      http:
+    - host: "craigbrookes.com"
```
Comment (philbrookes) [inline samples/echo-service/echo.yaml:42]: should this be changed back?

**Review by philbrookes — APPROVED** [review]

**Pull request merged**